drivers/azure/azure.go

```diff
@@ -325,13 +354,28 @@ func (d *Driver) Create() error {
 	if err := c.CreateStorageAccount(d.ctx, d.ResourceGroup, d.Location, defaultStorageType); err != nil {
 		return err
 	}
-	if err := d.generateSSHKey(d.ctx); err != nil {
-		return err
+
+	if d.OS != drivers.WINDOWS {
+		if err := d.generateSSHKey(d.ctx); err != nil {
+			return err
+		}
 	}
-	if err := c.CreateVirtualMachine(d.ResourceGroup, d.naming().VM(), d.Location, d.Size, d.ctx.AvailabilitySetID,
-		d.ctx.NetworkInterfaceID, d.BaseDriver.SSHUser, d.ctx.SSHPublicKey, d.Image, d.ctx.StorageAccount); err != nil {
-		return err
+
+	if d.OS == drivers.LINUX {
+		if err := c.CreateVirtualMachine(d.OS, d.ResourceGroup, d.naming().VM(), d.Location, d.Size, d.ctx.AvailabilitySetID,
+			d.ctx.NetworkInterfaceID, d.BaseDriver.SSHUser, "", d.ctx.SSHPublicKey, d.Image, d.ctx.StorageAccount); err != nil {
+			return err
+		}
+	} else if d.OS == drivers.WINDOWS {
+		if err := c.CreateVirtualMachine(d.OS, d.ResourceGroup, d.naming().VM(), d.Location, d.Size, d.ctx.AvailabilitySetID,
+			d.ctx.NetworkInterfaceID, d.BaseDriver.WinRMUser, d.BaseDriver.WinRMPassword, d.ctx.SSHPublicKey, d.Image, d.ctx.StorageAccount); err != nil {
+			return err
+		}
+		if err := c.CreateVirtualMachineExtension(d.OS, d.ResourceGroup, d.naming().VM(), d.Location); err != nil {
+			return err
+		}
```

Comment:
let's handle `else` here (and everhywhere else applicable). 

also for these stuff, `switch` statement is more idiomatic.